vernac/record.ml

```diff
@@ -583,17 +597,24 @@ let extract_record_data records =
   in
   ps, data
 
-type kind_class = NotClass | RecordClass | DefClass
-
-let kind_class =
-  let open Vernacexpr in
-  function Class true -> DefClass | Class false -> RecordClass
-  | Inductive_kw | CoInductive | Variant | Record | Structure -> NotClass
-
 let implicits_of_context ctx =
   List.map (fun name -> CAst.make (Some (name,true)))
     (List.rev (Anonymous :: (List.map RelDecl.get_name ctx)))
 
+(* deprecated in 8.16, to be removed at the end of the deprecation phase
+   (c.f., https://github.com/coq/coq/pull/15802 ) *)
+let warn_future_coercion_class_constructor =
+  CWarnings.create ~name:"future-coercion-class-constructor" ~category:"records"
+    ~default:CWarnings.AsError
+    Pp.(fun () -> str "'Class >' currently does nothing. Use 'Class' instead.")
+
+(* deprecated in 8.16, to be removed at the end of the deprecation phase
```

Comment:
This should be 8.17 right? Is the one above ok however?

Reply:
> This should be 8.17 right?

Right, the link to the PR was also wrong, fixed, thanks!

> Is the one above ok however?

Yes, that one remain 8.16